pkgs/development/libraries/science/math/magma/default.nix

```diff
@@ -1,48 +1,80 @@
-{ lib, stdenv, fetchurl, cmake, gfortran, ninja, cudaPackages, libpthreadstubs, lapack, blas }:
+{ lib
+, stdenv
+, fetchurl
+, cmake
+, ninja
+, gfortran
+, libpthreadstubs
+, lapack
+, blas
+, cudatoolkit
+, cudaFlags
+, hip
+, hipblas
+, hipsparse
+, openmp
+, useCUDA ? true
+, gpuTargets ? [ ]
+}:
 
-let
-  inherit (cudaPackages) cudatoolkit cudaFlags;
-in
-
-assert let majorIs = lib.versions.major cudatoolkit.version;
-       in majorIs == "9" || majorIs == "10" || majorIs == "11";
-
-let
+stdenv.mkDerivation (finalAttrs: {
+  pname = "magma";
   version = "2.6.2";
 
-in stdenv.mkDerivation {
-  pname = "magma";
-  inherit version;
   src = fetchurl {
-    url = "https://icl.cs.utk.edu/projectsfiles/magma/downloads/magma-${version}.tar.gz";
+    name = "magma-${finalAttrs.version}.tar.gz";
+    url = "https://icl.cs.utk.edu/projectsfiles/magma/downloads/magma-${finalAttrs.version}.tar.gz";
     hash = "sha256-dbVU2rAJA+LRC5cskT5Q5/iMvGLzrkMrWghsfk7aCnE=";
-    name = "magma-${version}.tar.gz";
   };
 
-  nativeBuildInputs = [ gfortran cmake ninja ];
+  nativeBuildInputs = [
+    cmake
+    ninja
+    gfortran
+  ];
 
-  buildInputs = [ cudatoolkit libpthreadstubs lapack blas ];
+  buildInputs = [
+    libpthreadstubs
+    lapack
+    blas
+  ] ++ lib.optionals useCUDA [
+    cudatoolkit
+  ] ++ lib.optionals (!useCUDA) [
```

Comment:
I prefer we use here a useROC flag and disable that by default to reduce the closure size.